optimizely/event_builder.py

```diff
@@ -139,58 +151,57 @@ class EventParams(object):
     CUSTOM = 'custom'
     ANONYMIZE_IP = 'anonymize_ip'
 
-  def _add_attributes(self, attributes):
-    """ Add attribute(s) information to the event.
+  def _get_attributes(self, attributes):
+    """ Get attribute(s) information.
 
     Args:
       attributes: Dict representing user attributes and values which need to be recorded.
+
+    Returns:
+      list Valid attributes for the user (if any). Otherwise, empty.
     """
 
-    visitor = self.params[self.EventParams.USERS][0]
-    visitor[self.EventParams.ATTRIBUTES] = []
+    params = []
 
     if not attributes:
-      return
+      return []
 
     for attribute_key in attributes.keys():
       attribute_value = attributes.get(attribute_key)
       # Omit falsy attribute values
       if attribute_value:
         attribute = self.config.get_attribute(attribute_key)
         if attribute:
-          visitor[self.EventParams.ATTRIBUTES].append({
+          params.append({
             self.EventParams.EVENT_ID: attribute.id,
             'key': attribute_key,
             'type': self.EventParams.CUSTOM,
             'value': attribute_value,
           })
 
-  def _add_source(self):
-    """ Add source information to the event. """
+    return params
 
-    self.params[self.EventParams.SOURCE_SDK_TYPE] = 'python-sdk'
-    self.params[self.EventParams.SOURCE_SDK_VERSION] = version.__version__
+  def _get_source(self):
+    """ Get source information.
 
-  def _add_user_id(self, user_id):
-    """ Add user ID to the event.
-
-    Args:
-      user_id: ID of the user.
+    Returns:
+      dict SDK type and version information.
```

Comment:
No need to set this like this. Rather just set it like the user ID is set i.e. in place.